### ISSUE TITLE
Add colorbar to image viewer

### ISSUE THREAD

**Issue by pllim** [glue-viz/glue#1087]
As part of the JWST DADF sprint, I am looking into how to add a colorbar to the default image viewer in Glue. I traced the `imshow` commands to `glue/viewers/image/layer_artist.py`, where `self._axes.imshow(...)` is always directly appended to a list of `artists`. It is not obvious where to insert a colorbar as indicated in Matplotlib tutorial:

```python
fig, ax = plt.subplots()
cax = ax.imshow(data, ...)
cbar = fig.colorbar(cax, ...)
```

I don't see why a colorbar has to be an optional button to get displayed. I think it should just *always* be displayed with an image is shown.

**Comment by astrofrog** [comment]
I agree this makes sense - I think this should be done inside the layer artist for images, for the data (not subsets). The layer artist has access to the imshow call and the axes, and you can get access to the figure from the axes. Is that enough for you to get started, or would you like some more information/examples?
